src/main/java/io/harness/cf/client/api/StorageRepository.java

```diff
@@ -98,6 +99,37 @@ public Optional<FeatureConfig[]> getCurrentAndPreviousFeatureConfig(@NonNull Str
     return Optional.empty();
   }
 
+  public FeatureSnapshot getFeatureSnapshot(@NonNull String identifier) {
+    Gson gson = new Gson();
+    final String flagKey = formatFlagKey(identifier);
+    final String pFlagKey = formatPrevFlagKey(identifier);
+
+    FeatureConfig pFlag = (FeatureConfig) cache.get(pFlagKey);
+    FeatureConfig cFlag = (FeatureConfig) cache.get(flagKey);
+
+    if (cFlag != null) {
```

Comment:
should there be a null check for pFlag as well here?  I'm assuming there is an assumption that if we dont have the current then we wont have the previous.

Reply:
if the previous flag does not exist we still want to package that to the snapshot even if its null.